jaxley/modules/base.py

```diff
@@ -1207,9 +1208,14 @@ def _get_state_names(self) -> Tuple[List, List]:
 
         Returns states seperated by comps and edges."""
         channel_states = [name for c in self.channels for name in c.channel_states]
-        synapse_states = [name for s in self.synapses for name in s.synapse_states]
+        synapse_states = [
+            name for s in self.synapses if s is not None for name in s.synapse_states
```

Comment:
the `not None` filter is necessary since `self.synapses` is populated with `[None]` under certain conditions

I'll admit I don't totally understand why it's necessary just yet, although I saw the comment about it preserving indexing :smile: